datasets/cord-19/cord-19.py

```diff
@@ -0,0 +1,134 @@
+  Jerry Sheehan and Zhihong Shen and B. Stilson and A. Wade and K. Wang and Christopher Wilhelm and Boya Xie and
+  D. Raymond and Daniel S. Weld and Oren Etzioni and Sebastian Kohlmeier},
+  journal={ArXiv},
+  year={2020}
+}
+"""
+
+# You can copy an official description
+_DESCRIPTION = """\
+The Covid-19 Open Research Dataset (CORD-19) is a growing resource of scientific papers on Covid-19 and related
+historical coronavirus research. CORD-19 is designed to facilitate the development of text mining and information
+retrieval systems over its rich collection of metadata and structured full text papers. Since its release, CORD-19
+has been downloaded over 75K times and has served as the basis of many Covid-19 text mining and discovery systems.
+
+The dataset itself isn't defining a specific task, but there is a Kaggle challenge that define 17 open research
+questions to be solved with the dataset: https://www.kaggle.com/allen-institute-for-ai/CORD-19-research-challenge/tasks
+"""
+
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URL = "https://ai2-semanticscholar-cord-19.s3-us-west-2.amazonaws.com/historical_releases/cord-19_2020-11-29.tar.gz"
+
+
+class Cord19(datasets.GeneratorBasedBuilder):
+    """The Covid-19 Open Research Dataset (CORD-19) is a growing resource of scientific papers on Covid-19."""
+
+    VERSION = datasets.Version("0.0.1")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="cord-19",
+            description="The whole dataset in a compressed file. Only title and "
+            "abstract of each article are loaded for now.",
+        ),
+        # datasets.BuilderConfig(name="second_domain", description="This part of my dataset covers a second domain"),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=datasets.Features(
+                {
+                    "cord_uid": datasets.Value("string"),
+                    "sha": datasets.Value("string"),
+                    "source_x": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "doi": datasets.Value("string"),
+                    "abstract": datasets.Value("string"),
+                    "publish_time": datasets.Value("string"),
+                    "authors": datasets.Value("string"),
+                    "journal": datasets.Value("string"),
+                    "url": datasets.Value("string"),
+                    # TODO "full_text" (complex nested json)
+                    # TODO "bib_entries" (complex nested json)
+                    # TODO "doc_embeddings" (separated file to be link by paper_id?)
```

Comment:
do you want to add them in this PR ?

Reply:
it's in progress but I can do another PR if you feel it's better

Reply:
It's ok to add them in this PR
We usually do one PR = one dataset

Reply:
That's what I had in mind (adding more configurations with the extra data).
I might not finished before the end of the hackathon though (and the dataset is perfectly usable with only metadata)

Reply:
Ok :) feel free to ping me when you update the dataset

Reply:
ping ;-)